components/brave_wallet/browser/zcash/zcash_wallet_service.cc

```diff
@@ -436,6 +406,78 @@ void ZCashWalletService::CreateTransaction(const std::string& chain_id,
   task->ScheduleWorkOnTask();
 }
 
+void ZCashWalletService::ShieldFunds(const std::string& chain_id,
+                                     mojom::AccountIdPtr account_id,
+                                     ShieldFundsCallback callback) {
+#if BUILDFLAG(ENABLE_ORCHARD)
+  if (IsZCashShieldedTransactionsEnabled()) {
+    CreateShieldAllTransaction(
+        chain_id, account_id.Clone(),
+        base::BindOnce(&ZCashWalletService::CreateShieldTransactionTaskDone,
+                       weak_ptr_factory_.GetWeakPtr(), chain_id,
+                       account_id.Clone(), std::move(callback)));
+  } else {
+    std::move(callback).Run(
+        std::nullopt,
+        l10n_util::GetStringUTF8(IDS_WALLET_METHOD_NOT_SUPPORTED_ERROR));
+  }
+#else
+  std::move(callback).Run(
+      std::nullopt,
+      l10n_util::GetStringUTF8(IDS_WALLET_METHOD_NOT_SUPPORTED_ERROR));
+#endif
+}
+
+#if BUILDFLAG(ENABLE_ORCHARD)
+
+void ZCashWalletService::CreateShieldAllTransaction(
+    const std::string& chain_id,
+    mojom::AccountIdPtr account_id,
+    CreateTransactionCallback callback) {
+  CHECK(IsZCashShieldedTransactionsEnabled());
+  if (shield_funds_task_) {
+    std::move(callback).Run(
+        base::unexpected(l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR)));
+    return;
+  }
```

Comment:
Why there is only one task possbile? Should that be per account?

Reply:
We will support only single account with shielding options in the MVP